pkg/providers/docker/docker.go

```diff
@@ -460,3 +467,11 @@ func (p *provider) UpgradeNeeded(_ context.Context, _, _ *cluster.Spec) (bool, e
 func (p *provider) RunPostControlPlaneCreation(ctx context.Context, clusterSpec *cluster.Spec, cluster *types.Cluster) error {
 	return nil
 }
+
+func prevWorkerNodeGroups(prevWorkerNodeGroups []v1alpha1.WorkerNodeGroupConfiguration) map[string]v1alpha1.WorkerNodeGroupConfiguration {
```

Comment:
nit: `buildMapForWorkerNodeGroupsByName` (or something else that tell us what the method is doing